### PR TITLE
[#1238] add healthCheck for isolation instances

### PR DESCRIPTION
**Pull request by chengyouling**

### CODE REVIEW

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java [hunk 1]
```diff
@@ -86,7 +96,11 @@ public List<ServiceInstance> filter(ServiceInstanceListSupplier supplier, List<S
       }
 
       synchronized (lock) {
-        isolatedInstances.remove(InstanceIDAdapter.instanceId(serviceInstance));
+        if (checkInstanceHealth(serviceInstance)) {
```
Comment (liubao68) [inline spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:99]: do not put check instance health in synchronized block

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java [hunk 2]
```diff
@@ -52,13 +61,14 @@ public InstanceIsolationServiceInstanceFilter(Environment environment,
   @SuppressWarnings("unused")
   public void onInstanceIsolatedEvent(InstanceIsolatedEvent event) {
     synchronized (lock) {
-      for (Iterator<String> iterator = isolatedInstances.keySet().iterator(); iterator.hasNext(); ) {
-        Long duration = isolatedInstances.get(iterator.next());
-        if (System.currentTimeMillis() - duration > 0) {
-          iterator.remove();
+      if (!env.getProperty(INSTAANCE_PING_ENABLED, boolean.class, true)) {
+        for (Iterator<String> iterator = isolatedInstances.keySet().iterator(); iterator.hasNext(); ) {
+          Long duration = isolatedInstances.get(iterator.next());
```
Comment (liubao68) [inline spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:66]: these code may be safely deleted.

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java [hunk 3]
```diff
@@ -33,6 +38,10 @@
 import com.huaweicloud.governance.event.InstanceIsolatedEvent;
 
 public class InstanceIsolationServiceInstanceFilter implements ServiceInstanceFilter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(InstanceIsolationServiceInstanceFilter.class);
+
+  private static final String INSTAANCE_PING_ENABLED = "spring.cloud.servicecomb.isolation.instance.ping.enabled";
```
Comment (liubao68) [inline spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:43]: default value set to false.

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java (hunk 1 above)
Comment (chengyouling) [reply spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java]: fixed

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java (hunk 2 above)
Comment (chengyouling) [reply spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java]: delete

File: spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java (hunk 3 above)
Comment (chengyouling) [reply spring-cloud-huawei-governance/src/main/java/com/huaweicloud/governance/adapters/loadbalancer/InstanceIsolationServiceInstanceFilter.java:42]: fixed

**Review by liubao68 — APPROVED** [review]

**Pull request merged**